adbe/adb_enhanced.py

```diff
@@ -922,9 +927,13 @@ def _get_all_packages(pm_cmd):
 
 
 def list_all_apps():
-    cmd = 'pm list packages'
-    packages = _get_all_packages(cmd)
-    print('\n'.join(packages))
+    # The command "pm list packages -e" does not return all installed and enabled Apps, the alternative is to use
```

Comment:
So what exactly does `pm list packages -e` miss? Disabled apps?

Reply:
`pm list packages -e` for some unknown reason does not list some enabled apps.  
`-e` should show all enabled Apps including system Apps  which is not the case as discussed in the thread: https://stackoverflow.com/questions/63416599/adb-shell-pm-list-packages-missing-some-packages

Reply:
Thanks. It is fine if we don't know the underlying cause. Add this StackOverflow thread in the code comments as well.

Reply:
I saw that the regex is reported as an anomaly by DeepSource Python, Can we replace it by this one `dumpsys package | grep "Package \[" | cut -d "\[" -f2 | cut -d "\]" -f1`.

Reply:
The new code is fine. Just confirm that `cut` command's arguments are the same on both Linux and Mac.

Reply:
It should be the same (https://superuser.com/questions/179368/how-does-macs-command-line-compare-to-linux), unfortunately I do not have a Mac to test. Can someone else confirm that please ?

Reply:
I got `cut: bad delimiter` as an error on Mac OS. How about we do this parsing in Python instead? That's much more portable as well.

Reply:
Yes sure, I will update it.